### PR TITLE
changes to authz data classes for json serialization

### PR DESCRIPTION
**Pull request by schandrika**
depends on https://github.com/eclipse-volttron/volttron-core/pull/209

### CODE REVIEW

**Comment by schandrika** [comment]
With changes in https://github.com/eclipse-volttron/volttron-core/pull/209 and changes in this PR I was able to test calls from control.py to authz method using below code
`
caps = authz.PubsubCapabilities()
caps.add_pubsub_capability(authz.PubsubCapability(topic_pattern="test_topic/subtopic", topic_access="publish"))
rpc_caps = authz.RPCCapabilities([authz.RPCCapability(resource="vip1.method1")])

result = opts.connection.server.vip.rpc.call(AUTH, "create_or_merge_agent_authz", identity="new_agent",
                                                                    pubsub_capabilities=caps,
                                                                    rpc_capabilities=rpc_caps).get()
`

File: src/volttron/services/auth/auth_service.py [hunk 1]
```diff
@@ -278,18 +293,30 @@ def create_or_merge_agent_group(self, *, name: str,
...
-    def create_or_merge_agent_authz(self, *, identity: str, protected_rpcs: set[authz.vipid_dot_rpc_method] = None,
-                                    roles: authz.AgentRoles = None, rpc_capabilities: authz.RPCCapabilities = None,
-                                    pubsub_capabilities: authz.PubsubCapabilities = None, comments: str = None,
+    def create_or_merge_agent_authz(self, *,
+                                    identity: str,
+                                    protected_rpcs: list[authz.vipid_dot_rpc_method] = None,
+                                    roles: authz.AgentRoles | dict = None,
+                                    rpc_capabilities: authz.RPCCapabilities | dict = None,
+                                    pubsub_capabilities: authz.PubsubCapabilities | dict = None,
+                                    comments: str = None,
```
Comment (kefeimo) [inline src/volttron/services/auth/auth_service.py:310]: Can you give an example on how to use rpc_capabilities as dict type? similar to the following:


`
caps = authz.PubsubCapabilities()
caps.add_pubsub_capability(authz.PubsubCapability(topic_pattern="test_topic/subtopic", topic_access="publish"))
rpc_caps = authz.RPCCapabilities([authz.RPCCapability(resource="vip1.method1")])

result = opts.connection.server.vip.rpc.call(AUTH, "create_or_merge_agent_authz", identity="new_agent",
pubsub_capabilities=caps,
rpc_capabilities=rpc_caps).get()
`

**Comment by kefeimo** [comment]
Mostly worked. Some good to have:
* rpc_capabilities: authz.RPCCapabilities | dict = None is not type safe. Bottom line fixed: authz.RPCCapabilities | dict | None= None
* Provide examples for usage per the dict type "pc_capabilities: authz.RPCCapabilities | dict".
* P.S.: pc_capabilities: authz.RPCCapabilities | dict -> pc_capabilities: authz.RPCCapabilities | Dict["Type-X", "Type-Y"]

**Review by kefeimo — APPROVED** [review]
Mostly worked. Some good to have:

rpc_capabilities: authz.RPCCapabilities | dict = None is not type safe. Bottom line fixed: authz.RPCCapabilities | dict | None= None
Provide examples as part of the doc-string, especially for usage per the dict type "pc_capabilities: authz.RPCCapabilities | dict".
P.S.: pc_capabilities: authz.RPCCapabilities | dict -> pc_capabilities: authz.RPCCapabilities | Dict["Type-X", "Type-Y"]

**Pull request merged**